File: Ozon/ng-ozon-client/src/app/models/transfer-entities/order.ts
import { Client } from './client';
import { OrderStates } from '../enums/order-states.enum';
import { Good } from './good';
import { IEntity } from '../Interfaces/entity.interface';

export class Order implements IEntity {

    constructor(
        public readonly id: number,
        public client: Client,
        public time: Date,
        public state: OrderStates,
        public goods: Good[]
    ) { }

}

File: Ozon/Models/TransferEntities/Order.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System;
using System.Collections.Generic;

namespace Ozon.Models.TransferEntities
{
    public class Order
    {
        public int Id { get; set; }

        public Client Client { get; set; }

        [JsonConverter(typeof(StringEnumConverter))]
        public OrderStates State { get; set; }

        public DateTime Time { get; set; }

        public List<Good> Goods { get; set; }
    }
}
